src/mmt_tcpip/lib/protocols/http2.c

```diff
@@ -267,3 +267,201 @@ int mmt_check_http2(ipacket_t *ipacket, unsigned proto_index) {
 		return 0;
 }
 
+int restore_http2_packet(uint8_t*data_out,const ipacket_t * packet,int proto_offset){
+ 	int header_length=0;
+ 	int offset_header_length = proto_offset -1;
+   	for (int i = offset_header_length; i < offset_header_length+4; i++) {
+      		  header_length = (header_length << 8) | data_out[i];
+   	 }
+   	 header_length=header_length & 0x00FFFFFF;
+ 	int new_length=0;
+   	 for (int i = offset_header_length; i < offset_header_length+4; i++) {
+      		  new_length = (new_length << 8) | packet->data[i];
+   	 }
+   	 new_length=new_length & 0x00FFFFFF;
+   	// printf("new_length %d, header_length %d\n",new_length,header_length);
+   	// printf("data_out %02X packet_data %02X \n",data_out[proto_offset],packet->data[proto_offset]);
+ 	memcpy((uint8_t*)data_out+proto_offset,packet->data+proto_offset,new_length+9);
+ 	return new_length-header_length;
+}
+int modify_get(uint8_t*data_out,int proto_offset){
+    	//Go to method field
+   	int header_length =0;
+      	// Get http2 protocol offset
+   	int offset_header_length = proto_offset -1;
+   	for (int i = offset_header_length; i < offset_header_length+4; i++) {
+      		  header_length = (header_length << 8) | data_out[i];
+   	 }
+   	header_length=header_length & 0x00FFFFFF;
+	uint8_t authority_amf[] = {0x41,0x8d,0x0b,0xa2,0x5c,0x2e,0x2e,0xdb,0xeb,0xba,0xcd,0xc7,0x80,0xf0,0x3f,0x7a,0x03,0x61,0x6d,0x66};
+	int authority_amf_length = sizeof(authority_amf) ;
+	//printf("[modify_get]authority_amf_length %d\n",authority_amf_length);
+	memcpy(data_out+proto_offset+9+header_length-2,authority_amf,authority_amf_length);
```

Comment:
Before doing `memcpy`, we need to check
- whether `data_out+proto_offset+9+header_length-2` is bigger than the capaiblity of `data_out`
-  whether `authority_amf_length` is not bigger than the capability of `data_out+proto_offset+9+header_length-2`